### PR TITLE
ICMSLST-1453 Add know bought from form to contacts list view. 

### PR DESCRIPTION
**Pull request by MattHolmes123**
Changes:
  - Remove field from main firearms forms.
  - Add field to Manage import contacts view.
  - Update submit application validation  to redirect to correct view.

### CODE REVIEW

File: web/domains/case/_import/fa_dfl/views.py [hunk 1]
```diff
@@ -341,19 +341,27 @@ def _get_dfl_errors(application: DFLApplication) -> ApplicationErrors:
...
-            url=reverse(
-                "import:fa:list-import-contacts", kwargs={"application_pk": application.pk}
-            ),
+    brought_from_errors = PageErrors(
+        page_name="Details of who bought from",
+        url=reverse("import:fa:manage-import-contacts", kwargs={"application_pk": application.pk}),
+    )
+
+    # User hasn't selected a value yet
+    if application.know_bought_from is None:
```
Comment (mikeylockwood) [inline web/domains/case/_import/fa_dfl/views.py:350]: Could you run the `ImportContactKnowBoughtFromForm` validation here or add a check for `brought_from is False and self.app.importcontact_set.exists()`?

Comment (MattHolmes123) [reply web/domains/case/_import/fa_dfl/views.py:350]: I did think that but I'd also have to do the the elif anyway.

File: web/domains/case/_import/fa/forms.py [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+        label="Do you know who you plan to buy/obtain these items from?",
+        widget=YesNoRadioSelectInline,
+    )
+
+    def __init__(self, *args, application: FaImportApplication, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.app = application
+
+    def clean_know_bought_from(self):
+        brought_from = self.cleaned_data["know_bought_from"]
```
Comment (davidwtbuxton) [inline web/domains/case/_import/fa/forms.py:21]: Speling conflict with `brought_from` and `know_bought_from`. Presumably should be either "bought" or "brought".

File: web/domains/case/_import/fa_oil/views.py [hunk 3]
```diff
@@ -143,19 +143,30 @@ def submit_oil(request: AuthenticatedHttpRequest, *, application_pk: int) -> Htt
...
             errors.add(page_errors)
 
-        if application.know_bought_from and not application.importcontact_set.exists():
-            page_errors = PageErrors(
-                page_name="Details of who bought from",
-                url=reverse(
-                    "import:fa:list-import-contacts", kwargs={"application_pk": application_pk}
-                ),
+        # Check know bought from
+        brought_from_errors = PageErrors(
```
Comment (davidwtbuxton) [inline web/domains/case/_import/fa_oil/views.py:147]: "Brought" vs "Bought", same as above.

**Review by davidwtbuxton — APPROVED** [review]
Comments on speling.

File: web/domains/case/_import/fa/forms.py (hunk 2 above)
Comment (MattHolmes123) [reply web/domains/case/_import/fa/forms.py:21]: haha my bad :)

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**